ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/SnapServer.java

```diff
@@ -608,11 +612,12 @@ public boolean test(final Pair<Bytes32, Bytes> pair) {
         return false;
       }
 
+      var hasNoRecords = recordLimit.get() == 0;
       var underRecordLimit = recordLimit.addAndGet(1) <= MAX_ENTRIES_PER_REQUEST;
       var underByteLimit =
           byteLimit.accumulateAndGet(0, (cur, __) -> cur + encodingSizeAccumulator.apply(pair))
               < maxResponseBytesFudgeFactor;
-      if (underRecordLimit && underByteLimit) {
+      if (hasNoRecords || (underRecordLimit && underByteLimit)) {
```

Comment:
Is this a "pathological" hive test case where the byteLimit is set too low to allow any records through?  

👍 This seems a reasonable fix if the spec says there must always be at least one record in the response, but we should add a comment about why we are doing this, because it isn't obvious why we would do this otherwise.

Reply:
There wasn't any "pathological" test for the responses that use this predicate, well none obvious anyway. The "pathological" test was for getByteCodes with a limit too small to allow any response. But since the spec does state we need to return at least one response I made the changes anyway to be compliant.